src/H5FDsubfiling/H5subfiling_common.c

```diff
@@ -648,21 +657,40 @@ H5_open_subfiles(const char *base_filename, uint64_t file_id, H5FD_subfiling_par
     if (!context_id_out)
         H5_SUBFILING_GOTO_ERROR(H5E_FILE, H5E_CANTOPENFILE, FAIL, "invalid subfiling context ID pointer");
 
-    /* Initialize new subfiling context ID based on configuration information */
-    if (init_subfiling(base_filename, file_id, subfiling_config, file_acc_flags, file_comm, &context_id) < 0)
-        H5_SUBFILING_GOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "couldn't initialize subfiling context");
+    /* Check if this file is already open */
+    H5E_BEGIN_TRY
+    {
+        context_id = H5_subfile_fid_to_context(file_id);
```

Comment:
Could you switch this routine to pass a pointer to the context_id as a parameter, to leave the return value solely for errors, instead of overloading the return value with both?   Another "code smell" that I've been gradually cleaning up when I get a chance, and it would be good to avoid spreading it further here.  :-)   (Ideally, we can get all internal routines away from the return value overloading)

Reply:
Sure, I can do that. Agreed that return values should generally be reserved for errors.

Reply:
Cool, thanks.  (See comment below about the H5E_BEGIN_TRY/H5E_END_TRY)